src/route.js

```diff
@@ -27,11 +29,24 @@ var Route = function (routes, uri) {
 Route.prototype = {
 
   /**
+  Attempt to match the uri, or call a notFound handler if nothing matches.
+
+  @method matchOrNotFound
+  @param {Object} routeLevel
+  **/
+  matchOrNotFound: function (routeLevel) {
+    this.match(routeLevel);
+
+    if (!this.fullMatch && this.notFound) {
```

Comment:
One thing you might want to do with Aviator is redirect a certain route, say, for sending all users to a permissions check or something. 

```
'/app': {
	target: appRouteTarget,
	'/': {
		target: permissionsRedirectRouteTarget
		'/*': 'checkPermissionsAndRedirect'
	},
	'/feature': {
		// ...
	}
}
```

If I read this correctly, the fullMatch check here means that a default 404 page would override such a redirect. Right?

```
'/app': {
	target: appRouteTarget,
	$notFound: 'defaultNotFoundHandler'
	'/': {
		target: permissionsRedirectRouteTarget
		'/*': 'checkPermissionsAndRedirect'
	},
	'/feature': {
		// ...
	}
}
```

We may not want to absolutely require a fullMatch thing. Or rather, I think we want a '/*' in the local route level counts to count as a fullMatch.